rococo/repositories/mongodb/mongodb_repository.py

```diff
@@ -0,0 +1,151 @@
+                elif isinstance(field_value, dict):
+                    field_value = str(field_value.get('entity_id')).replace('-', '')
+                elif isinstance(field_value, str):
+                    field_value = field_value.replace('-', '')
+            if isinstance(field_value, UUID):
+                field_value = str(field_value).replace('-', '')
+            if isinstance(field_value, datetime):
+                field_value = field_value.strftime('%Y-%m-%d %H:%M:%S')
+
+            data[field.name] = field_value
+        return data
+
+    def get_move_entity_to_audit_table_query(self, data):
+        instance = self.get_one(self.table_name, "entity_id", {'entity_id': data.get('entity_id')})
+        self.db[f"{self.table_name}_audit"].insert_one(instance)
+    
+    def get_save_query(self, data):
+        self.db[f"{self.table_name}"].get_one_and_update(
+            {'entity_id': data.get("entity_id")},
+            {'$set': data},
+        )
+    
+    def save(self, instance: VersionedModel, send_message: bool = False):
+        """Save func"""
+        data = self._process_data_before_save(instance)
+        with self.adapter:
+            self.adapter.run_transaction([lambda: self.get_move_entity_to_audit_table_query(data), lambda: self.get_save_query(data)])
+            if send_message:
+                # This assumes that the instance is now in post-saved state with all the new DB updates
+                message = json.dumps(instance.as_dict(convert_datetime_to_iso_string=True))
+                self.message_adapter.send_message(self.queue_name, message)
+        return instance
+
+    def _insert(self, data: Dict, collection_name: str):
+        self.db[collection_name].insert_one(data)
+
+    def _outdate(self, data: Dict, collection_name: str):
+        collection = self.db[collection_name]
+        return collection.find_one_and_update(
+            {'entity_id': data.get("entity_id"), 'latest': True, 'active': True},
+            {'$set': {'latest': False}},
+            upsert=False,
+            return_document=ReturnDocument.AFTER
+        )
+
+    def create(self, data: Dict, collection_name: str):
+        return self.update(data, collection_name)
+
+    def create_many(self, data: List[Dict], collection_name: str):
+        self.db[collection_name].insert_many(documents=data)
+
+    def update(self, data: Dict, collection_name: str):
+        with self.client.start_session() as session:
+            with session.start_transaction():
+                try:
+                    updated_doc = self._outdate(data, collection_name)
+                    if updated_doc:
+                        data.update({
+                            'previous_version': updated_doc.get('version'),
+                            'version': get_uuid4_hex(),
+                            'changed_on': datetime.utcnow().isoformat()
+                        })
+                    self._insert(data, collection_name)
```

Comment:
I think this code should set `changed_by_id` as well.  See how it is done in [VersionedModel.prepare_for_save](https://github.com/EcorRouge/rococo/blob/0e8d2e0cc655f110cadaa94a022b1a86751d94c6/rococo/models/versioned_model.py#L183)

Reply:
not all our models will be inherited from "VersionedModel" that will support prepare_for_save() so we can delay this after we have converted all of them.